packages/workspaces-extension/src/index.ts

```diff
@@ -46,13 +50,76 @@ const workspacesMenu: JupyterFrontEndPlugin<void> = {
   }
 };
 
+//Plugin id for workspaces indicator plugin
+const WORKSPACE_INDICATOR_PLUGIN_ID =
+  '@jupyterlab/workspaces-extension:indicator';
+//Command id for toggling workspace indicator
+const WORKSPACE_INDICATOR_COMMAND_ID = 'workspace-indicator:toggle';
+
+/**
+ * The extension providing workspace indicator at topbar
+ */
+const workspacesIndicator: JupyterFrontEndPlugin<void> = {
+  id: WORKSPACE_INDICATOR_PLUGIN_ID,
+  description: 'Adds a workspace indicator element at topbar',
+  requires: [
+    IWorkspacesModel,
+    IWorkspaceCommands,
+    IWindowResolver,
+    ITranslator,
+    ISettingRegistry
+  ],
+  autoStart: true,
+  activate: async (
+    app: JupyterFrontEnd,
+    model: IWorkspacesModel,
+    commands: IWorkspaceCommands,
+    resolver: IWindowResolver,
+    translator: ITranslator,
+    registry: ISettingRegistry
+  ) => {
+    const trans = translator.load('jupyterlab');
+    const openWorkspace = async (workspace: string) => {
+      await app.commands.execute(commands.open, { workspace: workspace });
+    };
+    const workspaceSelector = new WorkspaceSelectorWidget({
+      currentWorkspace: resolver.name,
+      identifiers: model.identifiers,
+      openWorkspace: openWorkspace,
+      model: model,
+      translator: translator
+    });
+    const isToggled = await registry.get(
+      WORKSPACE_INDICATOR_PLUGIN_ID,
+      'toggled'
+    );
+    workspaceSelector.setHidden(!isToggled.composite as boolean);
+    app.shell.add(workspaceSelector, 'top', { rank: 1000 });
+    app.commands.addCommand(WORKSPACE_INDICATOR_COMMAND_ID, {
+      label: trans.__('Show Workspace Indicator'),
+      isToggled: () => workspaceSelector.isVisible,
+      execute: () => {
+        workspaceSelector.setHidden(workspaceSelector.isVisible);
+        if (registry) {
```

Comment:
`registry` is always defined as per requirements, so this if is not needed, do I see it right?